### PR TITLE
Re-integrate the locked assets feature

### PR DESCRIPTION
**Pull request by dcoutts**
This is for the Byron compat era only. It is not included (and would not be effective) in the fully decentralised era.

This does not yet push the plumbing all the way through to the top.

Built on #601 and #602. It will need to be rebased before merging.

### CODE REVIEW

File: cardano-ledger/src/Cardano/Chain/UTxO/UTxOConfiguration.hs [hunk 1]
```diff
@@ -2,17 +2,34 @@
...
 import Cardano.Prelude
 
+import qualified Data.Set as Set
+
 import Cardano.Chain.Common.Address (Address)
+import Cardano.Chain.Common.Compact (CompactAddress, toCompactAddress)
+
 
--- | Additional configuration for ledger validation.
+-- | Additional configruation for ledger validation.
```
Comment (ruhatch) [inline cardano-ledger/src/Cardano/Chain/UTxO/UTxOConfiguration.hs]: Seems like an odd change ;)

File: cardano-ledger/src/Cardano/Chain/Block/Validation.hs [hunk 2]
```diff
@@ -419,6 +420,7 @@ data BodyEnvironment = BodyEnvironment
   , k                  :: !BlockCount
   , allowedDelegators  :: !(Set KeyHash)
   , protocolParameters :: !Update.ProtocolParameters
+  , utxoConfiguration  :: !UTxOConfiguration
```
Comment (ruhatch) [inline cardano-ledger/src/Cardano/Chain/Block/Validation.hs]: This is kind of pernickety, but could we put this as the second value in the record? I tried to match the order of the specs, apart from implementation details such as ProtocolMagic, which I put above the spec fields.

**Review by ruhatch — APPROVED** [review]
Just a small change to record ordering and a typo, otherwise great, thanks

File: cardano-ledger/src/Cardano/Chain/UTxO/UTxOConfiguration.hs (hunk 1 above)
Comment (dcoutts) [reply cardano-ledger/src/Cardano/Chain/UTxO/UTxOConfiguration.hs]: brain fart? correcting a correct spelling :-)

File: cardano-ledger/src/Cardano/Chain/Block/Validation.hs (hunk 2 above)
Comment (dcoutts) [reply cardano-ledger/src/Cardano/Chain/Block/Validation.hs]: np

**Comment by dcoutts** [comment]
Feedback addressed, but still pending merging of #601 and #602

**Comment by dcoutts** [comment]
Hmm, what's going on? github can see that I force pushed from b5d5373 to d141da5 but it has not updated this PR's view to reflect that fact.

**Comment by dcoutts** [comment]
OFFS. Closing and re-opening.

**Pull request closed without merging**